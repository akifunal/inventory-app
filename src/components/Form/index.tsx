type FormProps = {
	onSubmit?: () => void;
	children?: React.ReactNode;
};

const Form = ({ children, onSubmit }: FormProps) => {
	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();

		if (onSubmit) {
			onSubmit();
		}
	};

	return (
		<section id='forms' className='mt-4 p-2'>
			<form onSubmit={handleSubmit}>{children}</form>
		</section>
	);
};

export default Form;
