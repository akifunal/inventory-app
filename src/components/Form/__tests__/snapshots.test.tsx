import { render } from '@testing-library/react';
import Form from '../index';

describe('Input component', () => {
	it('should render and match the snapshot with default props', () => {
		// Arrange and Act
		const { asFragment } = render(<Form> Test </Form>);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});
});
