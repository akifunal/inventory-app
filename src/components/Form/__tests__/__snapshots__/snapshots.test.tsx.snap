// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render and match the snapshot with default props 1`] = `
<DocumentFragment>
  <section
    class="mt-4 p-2"
    id="forms"
  >
    <form>
       Test 
    </form>
  </section>
</DocumentFragment>
`;
